src/metabase/util.clj

```diff
@@ -44,28 +45,64 @@
 
 (defprotocol ITimestampCoercible
   "Coerce object to a `java.sql.Timestamp`."
-  (->Timestamp ^java.sql.Timestamp [this]
+  (coerce-to-timestamp ^java.sql.Timestamp [this] [this timezone-coercible]
     "Coerce this object to a `java.sql.Timestamp`. Strings are parsed as ISO-8601."))
 
+(defprotocol ITimeZoneCoercible
+  "Coerce object to `java.util.TimeZone`"
+  (coerce-to-timezone ^TimeZone [this]
+    "Coerce `this` to `java.util.TimeZone`"))
+
+(extend-protocol ITimeZoneCoercible
+  String
+  (coerce-to-timezone [this]
+    (TimeZone/getTimeZone this))
+  TimeZone
+  (coerce-to-timezone [this]
+    this)
+  DateTimeZone
+  (coerce-to-timezone [this]
+    (.toTimeZone this)))
+
 (declare str->date-time)
 
+(def ^TimeZone utc
+  "UTC TimeZone"
+  (coerce-to-timezone "UTC"))
+
 (extend-protocol ITimestampCoercible
-  nil       (->Timestamp [_]
-              nil)
-  Timestamp (->Timestamp [this]
-              this)
-  Date       (->Timestamp [this]
-               (Timestamp. (.getTime this)))
+  nil
+  (coerce-to-timestamp [_]
```

Comment:
style aside: this is harder to read than the previous formatting.

Reply:
+1 I am not confused by `(->Timestamp something)`